pkg/kubelet/kubelet_node_status.go

```diff
@@ -427,6 +427,13 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 	now := kl.clock.Now()
 	if utilfeature.DefaultFeatureGate.Enabled(features.NodeLease) && now.Before(kl.lastStatusReportTime.Add(kl.nodeStatusReportFrequency)) {
 		if !podCIDRChanged && !nodeStatusHasChanged(&originalNode.Status, &node.Status) {
+			// We must update the volumes in use even if no changes were made to the node
```

Comment:
When do we set the Node status to ready?  I wonder if the test case still has a race condition.  When the test [restarts kubelet](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/test/e2e/storage/utils/utils.go#L239), it:
1. Waits for the Node to become ready.
2. Immediately checks the mounts on the node.

Could the node be set to ready before we mark the volume as in use and cleanup the mounts?

Reply:
The node is set to ready on line 425 when kubelet decides that the node is ready. It is another setter ([ready condition setter](https://github.com/kubernetes/kubernetes/blob/92076f0d2e2409253d90c55c434035ef6b9bbcb6/pkg/kubelet/nodestatus/setters.go#L439)).

So if the node was ready before kubelet restarts, I think the node is still ready after kubelet restarts. Does this help?

Reply:
I discussed with @jingxu97. I missed one more step in the test case.  Before steps 1) and 2), the test also waits for the Pod API object [to be gone](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/test/e2e/storage/utils/utils.go#L240).  And that cannot happen until we've successfully unmounted the volume.  So there should not be a race condition in the test.

Reply:
How about adding a few words in the comments like this

"We must mark the volumes as ReportedInUse in volume manager's dsw even if no changes were made to the node status (no volumes were added or removed from the VolumeInUse list in node status). The reason is that in case of kubelet restarts, dsw is repopulated and volume ReportedInUse would be set to false first. If no status update is needed here, it means that the volumes are already in the VolumeInUse list from the node status. We need to make sure to set those volumes in dsw as ReportedInUse to true. 
// This MarkVolumeReportedInUse cannot be performed in dsw directly because it does not have the access to the node object. This also cannot be populated on node status manager init because the volume may not have been added to dsw at that time.

Reply:
Updated, ptal